packages/syft/src/syft/node/node.py

```diff
@@ -457,49 +460,101 @@ def root_client(self):
-            canonical_name_version_map.append((canonical_name, object_version))
+        klasses_to_be_migrated = []
 
-        # Track all object types from action store
-        action_object_types = [Action, ActionObject]
-        action_object_types.extend(ActionObject.__subclasses__())
-        for object_type in action_object_types:
+        for object_type in object_types:
             canonical_name = object_type.__canonical_name__
             object_version = object_type.__version__
-            canonical_name_version_map.append((canonical_name, object_version))
 
-        for canonical_name, current_version in canonical_name_version_map:
             migration_state = migration_state_service.get_state(context, canonical_name)
             if (
                 migration_state is not None
                 and migration_state.current_version != migration_state.latest_version
             ):
-                klasses_to_be_migrated.append(canonical_name)
+                klasses_to_be_migrated.append(object_type)
             else:
                 migration_state_service.register_migration_state(
                     context,
-                    current_version=current_version,
+                    current_version=object_version,
                     canonical_name=canonical_name,
                 )
 
         return klasses_to_be_migrated
 
+    def find_and_migrate_data(self):
+        # Track all object type that need migration for document store
+        context = AuthedServiceContext(
+            node=self,
+            credentials=self.verify_key,
+            role=ServiceRole.ADMIN,
+        )
+        document_store_object_types = [
+            partition.settings.object_type
+            for partition in self.document_store.partitions.values()
+        ]
+
+        object_pending_migration = self._find_klasses_pending_for_migration(
+            object_types=document_store_object_types
+        )
+
+        if object_pending_migration:
+            print(
+                "Object in Document Store that needs migration: ",
+                object_pending_migration,
+            )
+
+        # Migrate data for objects in document store
+        for object_type in object_pending_migration:
+            canonical_name = object_type.__canonical_name__
+            object_partition = self.document_store.partitions.get(canonical_name)
+            if object_partition is None:
+                continue
+
+            print(f"Migrating data for: {canonical_name} table.")
+            migration_status = object_partition.migrate_data(
+                to_klass=object_type, context=context
+            )
+            if migration_status.is_err():
+                raise Exception(
+                    f"Failed to migrate data for {canonical_name}. Error: {migration_status.err()}"
+                )
+
+        # Track all object types from action store
+        action_object_types = [Action, ActionObject]
```

Comment:
Since we don't have a stash for ActionObjects, hardcoding the Base classes here and tracking all the possible subclasses that can be stored in ActionStore.